extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/package.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.streaming
+
+import org.apache.spark.sql.DataFrameReader
+
+package object kinesis {
+
+  /**
+   * Add the `kinesis` method to DataFrameReader that allows people to read from Kinesis using
+   * the DataFileReader.
+   */
+  implicit class KinesisDataFrameReader(reader: DataFrameReader) {
```

Comment:
Is this pattern used in other data sources? This is cool for scala, but does not work for Java and so if we use this in examples, it would be confusing for Java users.